packages/tests/server/react/formData.test.tsx

```diff
@@ -0,0 +1,183 @@
+import { routerToServerAndClientNew } from '../___testHelpers';
+import { createQueryClient } from '../__queryClient';
+import { QueryClientProvider } from '@tanstack/react-query';
+import {
+  experimental_formDataLink,
+  httpBatchLink,
+  loggerLink,
+  splitLink,
+} from '@trpc/client';
+import { createTRPCReact } from '@trpc/react-query';
+import { CreateTRPCReactBase } from '@trpc/react-query/createTRPCReact';
+import { initTRPC } from '@trpc/server';
+import {
+  experimental_createMemoryUploadHandler,
+  experimental_isMultipartFormDataRequest,
+  experimental_parseMultipartFormData,
+  nodeHTTPFormDataContentTypeHandler,
+} from '@trpc/server/adapters/node-http/content-type/form-data';
+import { nodeHTTPJSONContentTypeHandler } from '@trpc/server/adapters/node-http/content-type/json';
+import { CreateHTTPContextOptions } from '@trpc/server/adapters/standalone';
+import { konn } from 'konn';
+import React, { ReactNode } from 'react';
+import { z } from 'zod';
+import { zfd } from 'zod-form-data';
+
+beforeAll(async () => {
+  const { FormData, File, Blob } = await import('node-fetch');
+  globalThis.FormData = FormData;
+  globalThis.File = File;
+  globalThis.Blob = Blob;
+});
+
+function formDataOrObject<T extends z.ZodRawShape>(input: T) {
+  return zfd.formData(input).or(z.object(input));
+}
+
+const ctx = konn()
+  .beforeEach(() => {
+    const t = initTRPC.context<CreateHTTPContextOptions>().create();
+
+    const appRouter = t.router({
+      polymorphic: t.procedure
+        .use(async (opts) => {
+          if (!experimental_isMultipartFormDataRequest(opts.ctx.req)) {
+            return opts.next();
+          }
+          const formData = await experimental_parseMultipartFormData(
+            opts.ctx.req,
+            experimental_createMemoryUploadHandler(),
+          );
```

Comment:
what's the use case?

Reply:
Thought it was cool to see if you can make a mutation that doesn't care if it gets `FormData` or a JSON-call